src/sync.ts

```diff
@@ -70,46 +75,53 @@ export async function trySyncWithCredentials(gitHubUser: GitHubUser) {
  * Note: no concurrent calls!
  */
 export async function sync(myGitHubUser: GitHubUser) {
+  // Get existing data first
+  const existingRepoStateByFullName = await getRepoStateByFullName();
+
   const prBlocksAtSyncStart = await getMyPrBlockList();
   const reviewRequestBlocksAtSyncStart = await getReviewRequestBlockList();
   const commentBlocksAtSyncStart = await getCommentBlockList();
 
   resetGitHubCallsCounter();
   const syncStartUnixMillis = Date.now();
   const settings = await getSettings();
-  const allReposIncludingDisabled = await getRepos();
-  const repos = allReposIncludingDisabled.filter((v) => v.monitoringEnabled);
-  const prevRepoStateByFullName = await getRepoStateByFullName();
-  const repoStateByFullNameBuilder = new Map<string, RepoState>();
-
-  // used purely as a starting point (#NOT_MATURE: what if user unsubscribed from them?):
-  const recentNotifications = await listRecentNotifications(
-    settings.getMinCommentCreateDate(),
-  );
 
   const user = (await getUser()).data;
   myGitHubUser.login = user.login;
   const userTeams = await listUserTeams();
   myGitHubUser.teamIds = userTeams.map((v) => v.id);
 
+  const allReposIncludingDisabled = await getRepos();
+  const repos = allReposIncludingDisabled.filter((v) => v.monitoringEnabled);
+  const repoStateByFullNameBuilder = new Map<string, RepoState>(
+    existingRepoStateByFullName,
+  ); // Preserve existing data!
+
+  // used purely as a starting point (#NOT_MATURE: what if user unsubscribed from them?):
+  const repoActivities = new Map<string, ListRepoEventsResponseDataType[0][]>();
+
+  for (const repo of repos) {
+    const activities = await listRepoActivity(
+      repo.owner,
+      repo.name,
+      settings.getMinCommentCreateDate(),
+      myGitHubUser,
+    );
+    repoActivities.set(repo.fullName(), activities);
+  }
+
   // It's probably better to do these GitHub requests in a sequential manner so that GitHub is not
   // tempted to block them even if user monitors many repos:
   for (const repo of repos) {
-    let repoStateBuilder = prevRepoStateByFullName.get(repo.fullName());
+    let repoStateBuilder = repoStateByFullNameBuilder.get(repo.fullName());
     if (!repoStateBuilder) {
       repoStateBuilder = new RepoState(repo.fullName());
     }
     repoStateByFullNameBuilder.set(repo.fullName(), repoStateBuilder);
 
-    const repoRecentNotifications = recentNotifications.filter(
```

Comment:
why do we need to change anything in the last two methods? Can we just do:

```
var activities = await listRepoActivity(
      repo.owner,
      repo.name,
      settings.getMinCommentCreateDate(),
      myGitHubUser,
    );
```

and pass those as a parameter and that's it?

Reply:
I mean all the changes in `export async function sync(myGitHubUser: GitHubUser)` and below